iFixit/react-commerce
# Use iFixit API to Create ProductList Pages if Data Missing In Strapi

## Overview
We currently just 404 on pages where the product list data isn't in Strapi, even if a wiki is on iFixit. This calls to the iFixit API and tries to fill in the parts pages if we are missing some data. This will make the breadcrumbs hierarchy on these pages no longer directly tied to the productList in Strapi, but it shouldn't matter too much because we will just source the missing Strapi info from iFixit anyway.  It also replaces the root page from the wiki hierarchy with the `All Parts` productList. 

## CR
The main thing I'm not super happy with is the `const` that consolidate the Strapi and iFixit data together, but I couldn't think of a way to make it better. Also is `wikiid` the best way to identify the Root Page.

## QA
So for your dev Strapi `Parts/Asus_Zenfone_2` should be a vaild page. However `Parts/Asus_Phone` shouldn't be valid for your dev server on master (Note: it is on production see https://es.ifixit.com/Parts/Asus_Phone ) but on this branch it should get  info from iFixit and not be a 404 page. Note that the breadcrumbs on es.ifixit.com for this page and your dev page on this branch should be different because one gets parent data from Strapi and the other from iFixit wikis. Also make sure that most parent breadcrumb should always be `All Parts`.

Also make sure that performance isn't hurt significantly by this pull on Strapi pages, and that performance is acceptable on the pages that have to fetch iFixit wikis.

Closes: #430
